daemon/restart.go

```diff
@@ -34,11 +35,16 @@ func (daemon *Daemon) ContainerRestart(name string, seconds *int) error {
 // gracefully stop, before forcefully terminating the container. If
 // given a negative duration, wait forever for a graceful stop.
 func (daemon *Daemon) containerRestart(container *container.Container, seconds int) error {
-	// Avoid unnecessarily unmounting and then directly mounting
+
+	// Optimization: avoid unnecessarily unmounting and then directly mounting
 	// the container when the container stops and then starts
-	// again
-	if err := daemon.Mount(container); err == nil {
-		defer daemon.Unmount(container)
+	// again. Doesn't work on Windows.
+	if runtime.GOOS != "windows" {
```

Comment:
Discussing this in the maintainers meeting; perhaps we should move this logic to https://github.com/moby/moby/blob/master/daemon/daemon.go#L1203-L1227 (which already has some windows exceptions)